### PR TITLE
Resolve issue 361

### PR DESCRIPTION
**Pull request by martinohanlon**
1 line code change to resolve https://github.com/mu-editor/mu/issues/361

### CODE REVIEW

**Comment by ntoll** [comment]
Oooh... thank you!

Umm... are there tests..? Don't worry if not, I'll add some when I've finished squashing the bug I'm looking into right now.

**Comment by martinohanlon** [comment]
I was going to make a note re tests. I couldn't find an obvious save test I could amended.

If you make a test could you make a reference to it, so I can see how you implemented it.

**Comment by ntoll** [comment]
All code branches have tests, so the original will be tested somewhere. You'll see the test when I merge your code. :-)

**Comment by martinohanlon** [comment]
I have found it. I will try and write the test.

**Comment by ntoll** [comment]
Perfect! Thank you @martinohanlon! :-)

File: mu/logic.py [hunk 1]
```diff
@@ -525,8 +525,8 @@ def save(self):
             tab.path = self._view.get_save_path(workspace)
         if tab.path:
             # The user specified a path to a file.
-            if not os.path.basename(tab.path).endswith('.py'):
-                # No extension given, default to .py
+            if os.path.splitext(os.path.basename(tab.path))[1] == '':
```
Comment (ZanderBrown) [inline mu/logic.py]: Call to `os.path.basename()` is unnecessary as `os.path.splitext` handles it internally

```py3
>>> import os
>>> os.path.splitext('/non.sense/to.test/this/thing.py.in')
('/non.sense/to.test/this/thing.py', '.in')
>>> os.path.splitext('/non.sense/to.test/this/thing')
('/non.sense/to.test/this/thing', '')
>>> 
```

**Comment by tjguk** [comment]
@ZanderBrown given that the PR is already merged, I think we could leave this under the heading of "redundant but harmless"?

**Pull request merged**